<a href="https://colab.research.google.com/github/igormurigor/colab/blob/main/%D0%9E%D1%88%D0%B8%D0%B1%D0%BA%D0%B8_%D0%BF%D0%BE_%D0%B0%D0%B4%D0%B0%D0%BF%D1%82%D0%B5%D1%80%D0%B0%D0%BC.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [8]:
from google.colab import drive
drive.mount('/content/gdrive')

import pandas as pd

df = pd.read_excel('/content/gdrive/My Drive/content/adapter.xlsx')  
display(df)

Drive already mounted at /content/gdrive; to attempt to forcibly remount, call drive.mount("/content/gdrive", force_remount=True).


adapter_id  B2E%  B2P%  ...  Orders_7  Orders_7,13  Orders_13
0           12    31    94  ...       426          429          3
1           11    18    93  ...        53           55          2
2           41    18    89  ...      4738         4933        195
3           30    17    89  ...      4717         5049        332
4           49    16    90  ...      3160         3222         62
5           50    14    88  ...      7262         7440        178
6           36    14    89  ...      2170         2251         81
7            3    12    82  ...      1178         1318        140
8           20     9    92  ...        90           95          5
9           52     7    87  ...      4760         4904        144
10          16     6    88  ...      1425         1543        118
11          21     0    87  ...        80           87          7
12          22     0    83  ...         4            5          1
13          24     0    78  ...       130          136          6
14          51     0    89  ...       294          300          6
15          37     0    88  ...        16           39         23
16          46     0    78  ...       113          117          4
17          48     0   100  ...         0           17         17

[18 rows x 12 columns]

**В период с 2021-10-11 по 2021-10-21  топ-5 адаптеров с наибольшим  процентом по метрике B2E : 12,11,41,30,49** . *Тип комбинации m_booking_session.multifr_type не указывался*




---

## 12 Адаптер

In [11]:
adapter_12 = pd.read_excel('/content/gdrive/My Drive/content/adapter_12.xlsx')  
display(adapter_12.head(10))

adapter_id  first_name  AirCompanyID  ... Orders_7  Orders_7,13  Orders_13
0          12  skyscanner           532  ...       76           77          1
1          12  skyscanner          1046  ...       32           33          1
2          12      yandex          1046  ...        2            2          0
3          12   aviasales           532  ...       80           80          0
4          12   aviasales           282  ...        7            7          0
5          12   aviasales           524  ...       53           53          0
6          12  skyscanner           524  ...       39           39          0
7          12      yandex           233  ...        3            3          0
8          12      yandex           532  ...        0            0          0
9          12  skyscanner           160  ...       25           25          0

[10 rows x 15 columns]

**В 12 адаптере по количеству ошибок со статусом 10 , выделялись такие АК как Nordwind,AZUR Air,Yamal Airlines,Royal Flight** 

In [13]:
adapter_12_errors = pd.read_excel('/content/gdrive/My Drive/content/adapter_12_errors.xlsx')  
display(adapter_12_errors.head(10))

quantity  adapter  ... reservation_error_booking                   error_name
0        20       12  ...                      4000                BookingCommon
1        16       12  ...                      4302  BookingPassengerCitizenship
2        15       12  ...                      4000                BookingCommon
3        12       12  ...                      4000                BookingCommon
4         9       12  ...                      4302  BookingPassengerCitizenship
5         9       12  ...                      4000                BookingCommon
6         9       12  ...                      4000                BookingCommon
7         7       12  ...                      4000                BookingCommon
8         7       12  ...                      4302  BookingPassengerCitizenship
9         6       12  ...                      4302  BookingPassengerCitizenship

[10 rows x 7 columns]

**В 12 адаптере много ошибок связаных с гражданством пассажира** 




---





# 11 Адаптер

In [15]:
adapter_11 = pd.read_excel('/content/gdrive/My Drive/content/adapter_11.xlsx')  
display(adapter_11.head(10))

adapter_id  first_name  AirCompanyID  ... Orders_7  Orders_7,13  Orders_13
0          11   aviasales           285  ...       23           24          1
1          11   aviasales           586  ...        0            0          0
2          11  skyscanner           941  ...        0            0          0
3          11      yandex           285  ...        0            1          1
4          11   aviasales           719  ...        0            0          0
5          11   aviasales           253  ...        0            0          0
6          11   aviasales            72  ...        0            0          0
7          11   aviasales           529  ...        0            0          0
8          11   aviasales           607  ...        1            1          0
9          11   aviasales           693  ...        9            9          0

[10 rows x 15 columns]

In [16]:
adapter_11_errors = pd.read_excel('/content/gdrive/My Drive/content/adapter_11_errors.xlsx')  
display(adapter_11_errors.head(10))

quantity  adapter  ... reservation_error_booking     error_name
0         4       11  ...                      4000  BookingCommon
1         3       11  ...                      4000  BookingCommon
2         2       11  ...                      4000  BookingCommon
3         1       11  ...                      4000  BookingCommon
4         1       11  ...                      4000  BookingCommon

[5 rows x 7 columns]

**Относительно небольшое количество заказов в 11 адаптере** 



---
# 41 Адаптер


In [17]:
adapter_41 = pd.read_excel('/content/gdrive/My Drive/content/adapter_41.xlsx')  
display(adapter_41.head(10))

adapter_id     first_name  AirCompanyID  ... Orders_7  Orders_7,13  Orders_13
0          41      aviasales           156  ...     3996         4155        159
1          41         yandex           156  ...      275          298         23
2          41     skyscanner           156  ...      350          357          7
3          41  aviasales-b2b           156  ...       75           77          2
4          41         google           156  ...        6            6          0
5          41            NaN           156  ...       25           28          3
6          41        admitad           156  ...       10           10          0
7          41          kayak           156  ...        0            0          0
8          41     biletyplus           156  ...        0            0          0
9          41         emails           156  ...        5            5          0

[10 rows x 15 columns]

In [18]:
adapter_41_errors = pd.read_excel('/content/gdrive/My Drive/content/adapter_41_errors.xlsx')  
display(adapter_41_errors.head(10))

quantity  adapter  ... reservation_error_booking     error_name
0       621       41  ...                    4000.0  BookingCommon
1       571       41  ...                    4000.0  BookingCommon
2       131       41  ...                    4000.0  BookingCommon
3        21       41  ...                    4000.0  BookingCommon
4         7       41  ...                       NaN            NaN
5         3       41  ...                    4000.0  BookingCommon
6         2       41  ...                    4000.0  BookingCommon
7         1       41  ...                       NaN            NaN
8         1       41  ...                       NaN            NaN

[9 rows x 7 columns]

**Яндекс лидер по ошибкам , все ошибки из разряда общих , но ошибки типа "Price Has change" не всегда попадают в правильный код ошики 4101 . У яндекса самые  популярные ErrorMessage : General air service Error;   One or more segments are not bookable;   на них приходится процентов 80% всех ошибок под кодом 4000**



---
# 30 Адаптер


In [19]:
adapter_30 = pd.read_excel('/content/gdrive/My Drive/content/adapter_30.xlsx')  
display(adapter_30.head(10))

adapter_id  first_name  AirCompanyID  ... Orders_7  Orders_7,13  Orders_13
0          30   aviasales           163  ...      660          685         25
1          30      yandex           532  ...       59           68          9
2          30      yandex           243  ...      106          112          6
3          30      yandex           163  ...       48           57          9
4          30      yandex           276  ...       36           40          4
5          30      yandex           238  ...        6           11          5
6          30  skyscanner           163  ...      515          527         12
7          30      yandex           459  ...       59           62          3
8          30   aviasales           532  ...      216          238         22
9          30   aviasales           276  ...      190          203         13

[10 rows x 15 columns]

In [20]:
adapter_30_errors = pd.read_excel('/content/gdrive/My Drive/content/adapter_30_errors.xlsx')  
display(adapter_30_errors.head(10))

quantity  adapter  ... reservation_error_booking           error_name
0       187       30  ...                    4000.0        BookingCommon
1       152       30  ...                    4101.0  BookingPriceChanged
2       129       30  ...                    4101.0  BookingPriceChanged
3        53       30  ...                    4101.0  BookingPriceChanged
4        52       30  ...                    4101.0  BookingPriceChanged
5        50       30  ...                    4000.0        BookingCommon
6        46       30  ...                    4000.0        BookingCommon
7        40       30  ...                    4000.0        BookingCommon
8        36       30  ...                    4101.0  BookingPriceChanged
9        35       30  ...                    4101.0  BookingPriceChanged

[10 rows x 7 columns]

В 30 Адаптере также видно , что Яндекса большой процент B2E% и обшибки связаны с изменением цены 



---

# 49 Адаптер

In [21]:
adapter_49 = pd.read_excel('/content/gdrive/My Drive/content/adapter_49.xlsx')  
display(adapter_49.head(10))

adapter_id     first_name  AirCompanyID  ... Orders_7  Orders_7,13  Orders_13
0          49      aviasales           802  ...     1039         1047          8
1          49      aviasales           757  ...      776          794         18
2          49         yandex           757  ...       40           40          0
3          49     skyscanner           757  ...      144          148          4
4          49     skyscanner           802  ...      117          117          0
5          49     skyscanner           238  ...       86           87          1
6          49  aviasales-b2b           757  ...      252          255          3
7          49  aviasales-b2b           802  ...       22           22          0
8          49     skyscanner           532  ...       58           58          0
9          49      aviasales           160  ...       85           88          3

[10 rows x 15 columns]

In [22]:
adapter_49_errors = pd.read_excel('/content/gdrive/My Drive/content/adapter_49_errors.xlsx')  
display(adapter_49_errors.head(10))

quantity  adapter  ... reservation_error_booking           error_name
0       221       49  ...                    4000.0        BookingCommon
1        80       49  ...                    4101.0  BookingPriceChanged
2        54       49  ...                    4101.0  BookingPriceChanged
3        50       49  ...                    4101.0  BookingPriceChanged
4        47       49  ...                    4000.0        BookingCommon
5        39       49  ...                    4000.0        BookingCommon
6        35       49  ...                    4000.0        BookingCommon
7        24       49  ...                    4000.0        BookingCommon
8        23       49  ...                    4000.0        BookingCommon
9        21       49  ...                    4000.0        BookingCommon

[10 rows x 7 columns]

В 49 Адаптере  Азимут и Победа  лидеры по количеству ошибок . У Победы 77 ошибок в yandex из них 54 сзязаны с изменением цены 

## Вывод:
- Самые поплуярные ошибки :
 1.	4302	BookingPassengerCitizenship 
 2. 4000 BookingCommon
 3. 4101 BookingPriceChanged

- В Метапоисковике Яндекс бОльшая часть ошибок связана с изменением цены 
- в 12 адаптере много ошибок связаны с гражданством пассажира 
